src/python/antares_xpansion/input_checker.py

```diff
@@ -348,82 +495,71 @@ def check_setting_option_value(option, value):
         :return: True if the option has the correct type, exits if the value has the wrong type
     """
 
-    options_legal_values = {
-        "method": ["benders_decomposition"],
-        "uc_type": ["expansion_accurate", "expansion_fast"],
-        "master": ["relaxed", "integer", "full_integer"],
-        "optimality_gap": None,
-        "relative_gap": None,
-        "cut_type": ["average", "yearly", "weekly"],
-        "week_selection": ["true", "false"],
-        "max_iteration": None,
-        "relaxed_optimality_gap": None,
-        "timelimit": None,
-        "yearly-weights": None,
-        "additional-constraints": None,
-        "log_level": None,
-    }
-    legal_values = options_legal_values.get(option)
+    if not check_setting_option_type(option, value):
+        flushed_print(
+            "check_settings : value %s for option %s has the wrong type!" % (value, option))
+        raise OptionTypeError
+
+    legal_values = options_types_and_legal_values.get(option)[1]
 
     skip_verif = ["yearly-weights", "additional-constraints", "solver"]
 
     if ((legal_values is not None) and (value in legal_values)) or (option in skip_verif):
         return True
 
     if (option == "optimality_gap") or (option == "relative_gap"):
-        try:
-            gap = float(value)
-            if float(value) >= 0:
-                return True
-            else:
-                raise ValueError
-        except ValueError:
-            print(
+        if float(value) >= 0:
+            return True
+        else:
+            flushed_print(
                 "Illegal value %s for option %s : only positive values are allowed"
                 % (value, option)
             )
+            raise GapValueError
+
     elif option == 'max_iteration':
         if value in ["+Inf", "+infini"]:
             return True
         else:
-            try:
-                max_iter = int(value)
-                if (max_iter == -1) or (max_iter > 0):
-                    return True
-            except ValueError:
+            max_iter = int(value)
+            if (max_iter == -1) or (max_iter > 0):
+                return True
+            else:
                 flushed_print('Illegal value %s for option %s : only -1 or positive values are allowed'
                               % (value, option))
-                sys.exit(1)
+                raise MaxIterValueError
     elif option == "relaxed_optimality_gap":
```

Comment:
Remove this check of a deprecated option

Reply:
removed